### PR TITLE
Make padding/border size floor node size

### PR DESCRIPTION
**Pull request by nicoburns**
# Objective

Fix node sizing behaviour where size is/should be determined by border.
(PR currently only contains failing tests)

## Context

https://github.com/bevyengine/bevy/pull/7795#issuecomment-1442350121

### CODE REVIEW

**Comment by nicoburns** [comment]
Some notes:

- The sum of border and padding should floor the size of the node
- Like any other min-content size this overrides both `size` and `max-size`
- Aspect-ratio applies to these sizes
-  Need to test for:
   - Leaf
   - Flexbox (container)
   - Flexbox (relative child)
   - Flexbox (absolute child)
   - Grid (container)
   - Grid (child)

**Comment by nicoburns** [comment]
I believe this is now ready if we ignore aspect-ratio. Which I'm inclined to do for the time being because I think that's quite a deep rabbit hole to go down and very unlikely combination of styles for people to be using.

File: src/compute/leaf.rs [hunk 1]
```diff
@@ -72,6 +72,12 @@ pub(crate) fn compute(
         }
     };
 
+    // Note: both horizontal and vertical percentage padding/borders are resolved against the container's inline size (i.e. width).
+    // This is not a bug, but is how CSS is specified (see: https://developer.mozilla.org/en-US/docs/Web/CSS/padding#values)
```
Comment (alice-i-cecile) [inline src/compute/leaf.rs:76]: This is an incredibly helpful comment, thank you.

File: src/compute/leaf.rs [hunk 2]
```diff
@@ -111,25 +119,33 @@ pub(crate) fn compute(
         };
 
         let size = node_size.unwrap_or(measured_size).maybe_clamp(node_min_size, node_max_size);
+        let size = size.maybe_max(padding_border.sum_axes().map(Some));
+        // let size = match sizing_mode {
```
Comment (alice-i-cecile) [inline src/compute/leaf.rs:123]: This commented out code either needs a comment of its own or to be removed.

File: src/compute/flexbox.rs [hunk 3]
```diff
@@ -607,88 +607,97 @@ fn determine_flex_base_size(
...
-            child.node,
-            child_known_dimensions,
-            constants.node_inner_size,
-            available_space,
-            SizingMode::ContentSize,
-        )
-        .main(constants.dir);
-    }
+        // Floor flex-basis by the padding_border_sum (floors inner_flex_basis at 0)
+        // This seems to be in violation of the spec which explicitly states that the content box should not be floored at
```
Comment (alice-i-cecile) [inline src/compute/flexbox.rs:677]: "should not be floored at" seems like a sentence fragment that's missing  a critical word.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
A couple of small quality issues, then this LGTM. We should resolve clippy failures from the new Rust version in their own PR.

File: src/compute/leaf.rs (hunk 2 above)
Comment (nicoburns) [reply src/compute/leaf.rs]: Removed

**Review by alice-i-cecile — APPROVED** [review]

**Review by Weibye — COMMENTED** [review]
Your comments are incredibly insightful and helpful, so keep that up :)

**Pull request merged**